cloud/deploy/bundle.go

```diff
@@ -0,0 +1,247 @@
+	if err != nil {
+		return err
+	}
+
+	// get the current deployment so we can check the deploy is valid
+	currentDeployment, err := getDeployment(c.Organization, input.DeploymentID, platformCoreClient)
+	if err != nil {
+		return err
+	}
+
+	// if CI/CD is enforced, check the subject can deploy
+	if currentDeployment.IsCicdEnforced && !canCiCdDeploy(c.Token) {
+		return fmt.Errorf(errCiCdEnforcementUpdate, currentDeployment.Name) //nolint
+	}
+
+	// check the deployment is enabled for DAG deploys
+	if !currentDeployment.IsDagDeployEnabled {
+		return fmt.Errorf(enableDagDeployMsg, input.DeploymentID) //nolint
+	}
+
+	// retrieve metadata about the local Git checkout. returns nil if not available
+	gitMetadata := retrieveLocalGitMetadata(input.BundlePath)
+
+	// initialize the deploy
+	deploy, err := createBundleDeploy(c.Organization, input, gitMetadata, coreClient)
+	if err != nil {
+		return err
+	}
+
+	// check we received an upload URL
+	if deploy.BundleUploadUrl == nil {
+		return errors.New("No bundle upload URL received from Astro")
+	}
+
+	// upload the bundle
+	tarballVersion, err := uploadBundle(config.WorkingPath, input.BundlePath, *deploy.BundleUploadUrl, false)
+	if err != nil {
+		return err
+	}
+
+	// finalize the deploy
+	_, err = finalizeBundleDeploy(c.Organization, deploy.Id, tarballVersion, input, coreClient)
+	if err != nil {
+		return err
+	}
+	fmt.Println("Successfully uploaded bundle with version " + tarballVersion + " to Astro.")
+
+	// if requested, wait for the deploy to finish by polling the deployment until it is healthy
+	if input.Wait {
+		err = deployment.HealthPoll(currentDeployment.Id, currentDeployment.WorkspaceId, dagOnlyDeploySleepTime, tickNum, timeoutNum, platformCoreClient)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func uploadBundle(tarDirPath, bundlePath, uploadURL string, prependBaseDir bool) (string, error) {
+	tarFilePath := filepath.Join(tarDirPath, "bundle.tar")
+	defer func() {
+		err := os.Remove(tarFilePath)
+		if err != nil {
+			fmt.Println("\nFailed to delete tar file: ", err.Error())
+			fmt.Println("\nPlease delete the tar file manually from path: " + tarFilePath)
+		}
+	}()
+
+	// Generate the bundle tar
+	err := fileutil.Tar(bundlePath, tarFilePath, prependBaseDir)
+	if err != nil {
+		return "", err
+	}
+
+	tarFile, err := os.Open(tarFilePath)
+	if err != nil {
+		return "", err
+	}
+	defer tarFile.Close()
+
+	versionID, err := azureUploader(uploadURL, tarFile)
+	if err != nil {
+		return "", err
+	}
+
+	return versionID, nil
+}
+
+func getDeployment(organizationID, deploymentID string, platformCoreClient astroplatformcore.CoreClient) (*astroplatformcore.Deployment, error) {
+	resp, err := platformCoreClient.GetDeploymentWithResponse(context.Background(), organizationID, deploymentID)
+	if err != nil {
+		return nil, err
+	}
+	err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.JSON200, nil
+}
```

Comment:
This exists almost identically as `CoreGetDeployment()` from ./cloud/deployment  package -- could we use that instead please?

Reply:
True, will use that.